src/main/java/dayum/dayumserver/application/member/MemberService.java

```diff
@@ -0,0 +1,102 @@
+package dayum.dayumserver.application.member;
+
+import static dayum.dayumserver.domain.member.Oauth2Provider.APPLE;
+import static dayum.dayumserver.domain.member.Oauth2Provider.NAVER;
+
+import dayum.dayumserver.application.member.dto.LoginRequest;
+import dayum.dayumserver.application.member.dto.LoginResponse;
+import dayum.dayumserver.application.member.dto.OAuthUserInfo;
+import dayum.dayumserver.client.s3.oauth2.naver.NaverOAuthClient;
+import dayum.dayumserver.domain.member.Member;
+import dayum.dayumserver.domain.member.MemberRepository;
+import dayum.dayumserver.domain.member.Oauth2Provider;
+import jakarta.transaction.Transactional;
+import java.time.LocalDateTime;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class MemberService {
+
+  private final MemberRepository memberRepository;
+  private final NaverOAuthClient naverOAuthClient;
+  private final JwtProvider jwtProvider;
+
+  public Member loginOrRegister(
+      OAuthUserInfo user,
+      String nickname,
+      String profileImage,
+      String bio,
+      Oauth2Provider provider) {
+
+    return memberRepository
+        .findByEmailAndProvider(user.email(), provider)
+        .orElseGet(
+            () ->
+                memberRepository.save(
+                    new Member(
+                        null,
+                        user.email(),
+                        user.name(),
+                        nickname,
+                        profileImage,
+                        provider,
+                        bio,
+                        false,
+                        null)));
+  }
+
+  public boolean isNicknameDuplicated(String nickname) {
+    return memberRepository.existsByNickname(nickname);
+  }
+
+  public LoginResponse login(LoginRequest request, Oauth2Provider provider) {
+    String accessToken = request.accessToken();
+
+    OAuthUserInfo userInfo =
+        switch (provider) {
+          case NAVER -> naverOAuthClient.getUserInfo(accessToken);
+          case APPLE -> throw new UnsupportedOperationException("Apple not implemented yet");
+        };
+
+    String nickname = Optional.ofNullable(request.nickname()).orElse(userInfo.name());
+    String profileImage =
+        Optional.ofNullable(request.profileImage()).orElse(userInfo.profileImage());
+    String bio = Optional.ofNullable(request.bio()).orElse("");
+
+    Member member = loginOrRegister(userInfo, nickname, profileImage, bio, provider);
+    String jwt = jwtProvider.createToken(member.id());
+    String refresh = jwtProvider.createRefreshToken(member.id());
```

Comment:
```suggestion
    String accessToken = jwtProvider.createToken(member.id());
    String refreshToken = jwtProvider.createRefreshToken(member.id());
```

변수명이 겹치는데 56L 의 변수명을 "OAuth2Provider 측의 토큰" 이라는 의미로 변경하는게좋아보입니다.